backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/invoice/impl/JSONInvoiceInfoResponse.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * #%L
+ * de.metas.business.rest-api-impl
+ * %%
+ * Copyright (C) 2020 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.rest_api.invoice.impl;
+
+import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import de.metas.currency.CurrencyCode;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Singular;
+import lombok.Value;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Builder
+@Value
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE, isGetterVisibility = JsonAutoDetect.Visibility.NONE, setterVisibility = JsonAutoDetect.Visibility.NONE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(builder = JSONInvoiceInfoResponse.JSONInvoiceInfoResponseBuilder.class)
+public class JSONInvoiceInfoResponse
+{
+	@Singular
+	@NonNull
+	List<JSONInvoiceLineInfo> lineInfos;
+
+	@JsonIgnoreProperties(ignoreUnknown = true)
+	@JsonPOJOBuilder(withPrefix = "")
+	static class JSONInvoiceInfoResponseBuilder
+	{
+	}
+}
+
+
+@Builder
+@Value
+@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY, getterVisibility = JsonAutoDetect.Visibility.NONE, isGetterVisibility = JsonAutoDetect.Visibility.NONE, setterVisibility = JsonAutoDetect.Visibility.NONE)
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonDeserialize(builder = JSONInvoiceLineInfo.JSONInvoiceLineInfoBuilder.class)
+class JSONInvoiceLineInfo
+{
+	int line_number;
+	String product_name;
+	BigDecimal qty_invoiced;
+	BigDecimal price;
+	BigDecimal tax_rate;
+	BigDecimal line_net_amt;
+	CurrencyCode currency;
```

Comment:
is there any reason for this naming convention?
i would prefer camel case

Reply:
In order not to add the annotation `@JsonProperty` for all of them.

Reply:
u can have camel case and not having to annotate the fields with JsonProperty.
Is there some standard about how to fields shall look in the json?

Reply:
renamed to camelCase